x-pack/plugins/security_solution/server/lib/telemetry/endpoint_task.ts

```diff
@@ -122,8 +133,15 @@ export class TelemetryEndpointTask {
       return 0;
     }
 
-    const endpointData = await this.fetchEndpointData();
+    const endpointData = await this.fetchEndpointData(executeFrom, executeTo);
 
+    /** STAGE 1 - Fetch Endpoint Agent Metrics
+     *
+     * Reads Endpoint Agent metrics out of the `.ds-metrics-endpoint.metrics` data stream
+     * and buckets them by Endpoint Agent id and sorts by the top hit. The EP agent will
+     * report its metrics once per day OR every time a policy change has occured. If
+     * a metric document(s) exists for an EP agent we map to fleet agent and policy
+     */
     const { body: endpointMetricsResponse } = (endpointData.endpointMetrics as unknown) as {
```

Comment:
Just responding to your question from slack here, I know you didn't change this but yeah if you have time to make a few changes here that'd be great.

I believe it's possible for `endpointMetricsResponse` to be `undefined based on the return value of the `this.fetchEndpointData()` call. Below I think we can change the `if` to `if (endpointMetricsResponse?.aggregation` to address that

I think if we could remove the cast to `unknown` that'd be awesome. I _think_ we can achieve that by adding something like this:

```
interface MetricsType {
  key: string;
  doc_count: number;
  latest_metrics: EndpointMetricHits;
}

const endpointAgentsAggs = endpointMetricsResponse.aggregations
      .endpoint_agents as AggregationsMultiBucketAggregate<MetricsType>;

    const endpointMetrics = endpointAgentsAggs.buckets.map((epMetrics) => {
      return {
        endpoint_agent: epMetrics.latest_metrics.hits.hits[0]._source.agent.id,
        endpoint_version: epMetrics.latest_metrics.hits.hits[0]._source.agent.version,
        endpoint_metrics: epMetrics.latest_metrics.hits.hits[0]._source,
      };
```

As a side note it looks like `EndpointMetricsAggregation` might need `total` I think that is already typed as part of the `SearchResponse` type, so we might be able to get rid of that part.

```
export interface EndpointMetricsAggregation {
  hits: {
    total: { value: number };
  };
```